packages/hardhat-chai-matchers/src/bigNumber.ts

```diff
@@ -0,0 +1,329 @@
+      } else {
+        throw new Error(`Unknown comparison operation ${method}`);
+      }
+    }
+    if (chaiUtils.flag(this, "doLength") && isBigNumber(actualArg)) {
+      const sizeOrLength =
+        expectedFlag instanceof Map || expectedFlag instanceof Set
+          ? "size"
+          : "length";
+      if (expectedFlag[sizeOrLength] === undefined) {
+        _super.apply(this, args);
+        return;
+      }
+      const expected = normalize(expectedFlag[sizeOrLength]);
+      const actual = normalize(actualArg);
+      this.assert(
+        compare(functionName, expected, actual),
+        `expected #{this} to have a ${sizeOrLength} ${readableName.replace(
+          "be ",
+          ""
+        )} ${actual.toString()} but got ${expected}`,
+        `expected #{this} to have a ${sizeOrLength} ${readableNegativeName} ${actual.toString()}`,
+        expected,
+        actual
+      );
+    } else if (isBigNumber(expectedFlag) || isBigNumber(actualArg)) {
+      const expected = normalize(expectedFlag);
+      const actual = normalize(actualArg);
+      this.assert(
+        compare(functionName, expected, actual),
+        `expected ${expected} to ${readableName} ${actual}. The numerical values of the given "${typestr(
+          expectedFlag
+        ).toString()}" and "${typestr(
+          actualArg
+        ).toString()}" inputs were compared, and they differed.`,
+        `expected ${expected} to ${readableNegativeName} ${actual}. The numerical values of the given "${typestr(
+          expectedFlag
+        ).toString()}" and "${typestr(
+          actualArg
+        ).toString()}" inputs were compared, and they differed.`,
+        actual.toString(),
+        expected.toString()
+      );
+    } else {
+      _super.apply(this, args);
+    }
+  };
+}
+
+function overrideWithin(utils: Chai.ChaiUtils) {
+  return (_super: (...args: any[]) => any) =>
+    overwriteBigNumberWithin(_super, utils);
+}
+
+function overwriteBigNumberWithin(
+  _super: (...args: any[]) => any,
+  chaiUtils: Chai.ChaiUtils
+) {
+  return function (this: Chai.AssertionStatic, ...args: any[]) {
+    const [startArg, finishArg] = args;
+    const expectedFlag = chaiUtils.flag(this, "object");
+    if (
+      isBigNumber(expectedFlag) ||
+      isBigNumber(startArg) ||
+      isBigNumber(finishArg)
+    ) {
+      const expected = normalize(expectedFlag);
+      const start = normalize(startArg);
+      const finish = normalize(finishArg);
+      this.assert(
+        start <= expected && expected <= finish,
+        `expected ${expected} to be within ${start}..${finish}`,
+        `expected ${expected} to not be within ${start}..${finish}`,
+        expected,
+        [start, finish]
+      );
+    } else {
+      _super.apply(this, args);
+    }
+  };
+}
+
+function overrideCloseTo(utils: Chai.ChaiUtils) {
+  return (_super: (...args: any[]) => any) =>
+    overwriteBigNumberCloseTo(_super, utils);
+}
+
+function overwriteBigNumberCloseTo(
```

Comment:
Doing `expect(1).to.be.closeTo(2)` throws this error:

```
AssertionError: the arguments to closeTo or approximately must be numbers, and a delta is required
```

But doing `expect(1n).to.be.closeTo(2)` throws this:

```
Error: cannot convert undefined to BigInt
```

Not a huge deal, really, but I would fix it if it's not a lot of work.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/2428/commits/0e93f3537044051aa362c9974c768ca78aa50c6c